### PR TITLE
adapter: Remove old migrations

### PR DESCRIPTION
**Pull request by ParkMyCar**
### Motivation

* This PR refactors existing code.
    Removes old stash migrations, and bumps the minimum version we test in the upgrade tests to v0.52, which has been released to all customers. Note: Today (May 10th) all customer environments will be upgraded to v0.53.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - N/a

### CODE REVIEW

File: test/upgrade/mzcompose.py
Comment (jkosh44) [inline test/upgrade/mzcompose.py:80]: Make sure to get someone from QA to look at this.

**Review by jkosh44 — APPROVED** [review]

File: test/upgrade/mzcompose.py
Comment (ParkMyCar) [reply test/upgrade/mzcompose.py:80]: cc @def- what do you think about this?

Comment (def-) [reply test/upgrade/mzcompose.py:80]: All customers being on 0.52.0 already makes this seem reasonable to me.
I haven't actually touched these legacy upgrade tests yet, but based on what I can see this should still work fine, and since chain upgrades were not tested, shouldn't be affected.

If this is not urgent you could wait until tomorrow to see if @philip-stoev has a different opinion, he's probably the expert on this.

**Review by def- — APPROVED** [review]

**Comment by def-** [comment]
One question I still have is this: Are all objects actually migrated already? Do we migrate them immediately on upgrade or is it possible that customers have an upgraded system with still-unmigrated objects?

**Comment by philip-stoev** [comment]
@def- after this is merged you will start seeing upgrade failures in Nightly ... I will try to refactor the respective Buildkite steps sometime next week after I get a sign-off from @benesch on https://www.notion.so/materialize/Product-Upgrade-Paths-when-Testing-b6a3caf3a8424acfb440d455dbcf0e9a

For the record, I think aggressively removing all migrations except the really latest latest ever is risky going forward and we should re-evaluate our policy around that . I understand the desire to remove old code, however going forward I expect we will have more and more cases where there straight-version-by-version upgrades can not be guaranteed , either because of bugs in the orchestration procedures, unforeseen complicated outages, or planned features such as suspending of environments and/or backup+restore.

File: test/upgrade/mzcompose.py
Comment (philip-stoev) [reply test/upgrade/mzcompose.py:80]: This particular option here is/was not used since CI runs this test with `--most-recent=2`, which overrides the min-version.

**Pull request merged**
